## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 488393`
Commit: `a27e93efdf2cea556856e05744a6963a54b5d521`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>md2pdf</li>
    <li>md2pdf.dist</li>
    <li>python313Packages.md2pdf</li>
    <li>python313Packages.md2pdf.dist</li>
    <li>python314Packages.md2pdf</li>
    <li>python314Packages.md2pdf.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>md2pdf</summary>
<pre>        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig
------------------------------ Captured log call -------------------------------
WARNING  weasyprint:text.py:190 .notdef glyph rendered for Unicode string unsupported by fonts: &quot;👋&quot; (U+1F44B)
=========================== short test summary info ============================
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_and_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey Jane 👋\nID: 2&#x27;
========================= 3 failed, 24 passed in 2.44s =========================</pre>
</details>
<details>
<summary>python313Packages.md2pdf</summary>
<pre>        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig
------------------------------ Captured log call -------------------------------
WARNING  weasyprint:text.py:190 .notdef glyph rendered for Unicode string unsupported by fonts: &quot;👋&quot; (U+1F44B)
=========================== short test summary info ============================
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_and_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey Jane 👋\nID: 2&#x27;
========================= 3 failed, 24 passed in 2.45s =========================</pre>
</details>
<details>
<summary>python314Packages.md2pdf</summary>
<pre>        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
        /var/cache/fontconfig
        /homeless-shelter/.cache/fontconfig
------------------------------ Captured log call -------------------------------
WARNING  weasyprint:text.py:190 .notdef glyph rendered for Unicode string unsupported by fonts: &quot;👋&quot; (U+1F44B)
=========================== short test summary info ============================
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey John 👋\nID: 2&#x27;
FAILED tests/test_core.py::test_generate_pdf_with_jinja_frontmatter_and_context_input - AssertionError: assert &#x27;\x00H\x00e\x...00:\x00 \x002&#x27; == &#x27;Hey Jane 👋\nID: 2&#x27;
========================= 3 failed, 24 passed in 2.34s =========================</pre>
</details>
</details>

--- a/pkgs/by-name/md/md2pdf/package.nix
+++ b/pkgs/by-name/md/md2pdf/package.nix
@@ -1,0 +1,9 @@
+{
+  python3Packages,
+}:
+
+python3Packages.toPythonApplication (
+  python3Packages.md2pdf.overridePythonAttrs (oldAttrs: {
+    dependencies = oldAttrs.dependencies ++ oldAttrs.optional-dependencies.cli;
+  })
+)

--- a/pkgs/development/python-modules/md2pdf/default.nix
+++ b/pkgs/development/python-modules/md2pdf/default.nix
@@ -1,6 +1,5 @@
 {
   buildPythonPackage,
-  click,
   fetchFromGitHub,
   hatchling,
   jinja2,
@@ -8,28 +7,30 @@
   markdown,
   pygments,
   pymdown-extensions,
+  pypdf,
   pytest-cov-stub,
   pytestCheckHook,
   python-frontmatter,
+  typer,
+  watchfiles,
   weasyprint,
 }:
 
 buildPythonPackage rec {
   pname = "md2pdf";
-  version = "2.1.0";
+  version = "3.1.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "jmaupetit";
     repo = "md2pdf";
     tag = "v${version}";
-    hash = "sha256-oVWUoWIS9GmkNPbJg90diT0jEgOULySSNxNdg95T2Vs=";
+    hash = "sha256-ksccl9K0o0mZleyLe1K1ob78W2MKZksTFtu6/dZUWeg=";
   };
 
   build-system = [ hatchling ];
 
   dependencies = [
-    click
     jinja2
     markdown
     pygments
@@ -38,16 +39,36 @@ buildPythonPackage rec {
     weasyprint
   ];
 
+  optional-dependencies = {
+    cli = [
+      typer
+      watchfiles
+    ];
+    latex = [
+      # FIXME package markdown-latex
+    ];
+  };
+
   pythonImportsCheck = [ "md2pdf" ];
 
   nativeCheckInputs = [
+    pypdf
     pytest-cov-stub
     pytestCheckHook
-  ];
+  ]
+  ++ lib.concatAttrValues optional-dependencies;
 
   preCheck = ''
     export PATH="$out/bin:$PATH"
   '';
+
+  disabledTests = [
+    # AssertionError caused by
+    #     glyph rendered for Unicode string unsupported by fonts: "👋" (U+1F44B)
+    "test_generate_pdf_with_jinja_context_input"
+    "test_generate_pdf_with_jinja_frontmatter_and_context_input"
+    "test_generate_pdf_with_jinja_frontmatter_input"
+  ];
 
   meta = {
     changelog = "https://github.com/jmaupetit/md2pdf/blob/${src.tag}/CHANGELOG.md";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2820,8 +2820,6 @@ with pkgs;
 
   md2gemini = with python3.pkgs; toPythonApplication md2gemini;
 
-  md2pdf = with python3Packages; toPythonApplication md2pdf;
-
   mdcat = callPackage ../tools/text/mdcat {
     inherit (python3Packages) ansi2html;
   };

